interfaces/builtin/download_manager.go

```diff
@@ -0,0 +1,284 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+    "bytes"
+    "fmt"
+
+    "github.com/snapcore/snapd/interfaces"
+)
+
+/* The methods: allowGSMDownload, createMmsDownload, exit and setDefaultThrottle
+   are deliberately left out of this profile due to their privileged nature. */
+var downloadConnectedPlugAppArmor = []byte(`
+# Description: Can access the download manager.
+# Usage: common
+
+#include <abstractions/dbus-strict>
```

Comment:
I just noticed this. Why do you need access to the system bus? All your apparmor rules are for the session bus.